src/_guides/language/language-tour_migrated.md

```diff
@@ -110,22 +110,26 @@ mind:
 -   Everything you can place in a variable is an *object*, and every
     object is an instance of a *class*. Even numbers, functions, and
     `null` are objects.
-    All objects except `null` inherit from the [Object][] class.
+    With the exception of `null` (if you enable [sound null safety][ns]),
+    all objects inherit from the [Object][] class.
+
+    {{site.alert.version-note}}
+      [Null safety][ns] was introduced in Dart 2.12.
+    {{site.alert.end}}
 
 -   Although Dart is strongly typed, type annotations are optional
     because Dart can infer types. In the code above, `number`
     is inferred to be of type `int`.
 
--   If you enable [sound null safety][ns],
+-   If you enable [null safety][ns],
     variables can’t contain `null` unless you say they can.
     You can make a variable nullable by
     putting a question mark (`?`) at the end of its type.
     For example, a variable of type `int?` might be an integer,
     or it might be `null`.
-
-    {{site.alert.version-note}}
-      [Null safety][ns] was introduced in Dart 2.12.
-    {{site.alert.end}}
+    If you _know_ an expression is non-null but Dart disagrees,
+    you can add `!` to make it be treated as non-null.
```

Comment:
"be treated as" is vague about what happens if it does evaluate to null. Maybe: "you can add `!` to assert that it won't be null". And then clarify that `!` throws an exception otherwise?